feat: add ZhiPuAI NLP adapter support

feuyeux opened the pull request:
## Description
This PR adds support for ZhiPuAI as a new NLP adapter, enabling users to leverage ZhiPuAI's language models within Parlant.

## Changes
- Added `zhipu_service.py` with full ZhiPuAI adapter implementation
- Added comprehensive test coverage in `test_zhipu_service.py`
- Updated `pyproject.toml` with ZhiPuAI dependencies

## Testing
- All new tests pass
- Integration tested with ZhiPuAI API

mc-dorzo commented on pyproject.toml:
```diff
@@ -135,6 +136,9 @@ fireworks = ["fireworks-ai"]
 
 mistral = ["mistralai"]
 
+zhipu = ["zhipuai", "tiktoken"]
```
`tiktoken` is not optional; therefore, it does not need to be mentioned here

mc-dorzo commented on src/parlant/adapters/nlp/zhipu_service.py:
```diff
@@ -0,0 +1,605 @@
+                extra={},
+            )
+
+            return SchematicGenerationResult(
```
Please take a look at other NLP services  adapters and add `record_llm_metrics`

mc-dorzo commented on src/parlant/adapters/nlp/zhipu_service.py:
```diff
@@ -0,0 +1,605 @@
+        return 128 * 1024
+
+
+class ZhipuEmbedder(Embedder):
```
Please inherit from BaseEmbedder

feuyeux commented:
Hi @mc-dorzo,

Thanks for the detailed review! I've addressed all 6 comments:

**1. pyproject.toml (line 139)** 
- Removed `tiktoken` from the `zhipu` extras since it's already a core dependency

**2. zhipu_service.py (line 251)** 
- Added `record_llm_metrics` call in the `_do_generate` method, following the OpenAI service pattern

**3. zhipu_service.py (line 477)** 
- Implemented `self._hist_moderation_request_duration` histogram tracking in `do_moderate`
- Refactored to use the same pattern as OpenAI service with a wrapper method and internal `_do_moderate`

**4-6. Inheritance comments** 
- Confirmed that `ZhipuSchematicGenerator` already inherits from `BaseSchematicGenerator[T]` (line 105)
- Confirmed that `ZhipuEmbedder` already inherits from `BaseEmbedder` (line 334)
- Both implementations follow the OpenAI service structure

All changes have been tested and pass diagnostics. Ready for another review!

kichanyurd commented:
@mc-dorzo is this ready to merge as far as you're concerned?

mc-dorzo commented:
@feuyeux just merged it — thanks, mate! Looking forward to working with you more!
